src/edu/psu/sweng500/team8/play/GameSession.java

```diff
@@ -56,39 +58,72 @@ public Board getGameBoard() {
 	 * @param cell
 	 * @param number
 	 */
-	public void enterNumber(Cell currentCell, int number) {
-		
-		/* keep track of the last action*/
-		SudokuAction sudokuAction = new SudokuAction(new CellGrid(board.getCellGrid()));
-		
-		boolean isDelete = false;
-		
-		if (number == 0) {
-			isDelete = true;			
-		}
-		
-		if (isDelete) {
-			currentCell.clearNumber();
-		} else {
-			currentCell.setNumber(number);	
-		}
-		
-		/* FIXME - PencilMarkManager updating/wiping out PencilMark number matching entered number */
-		if (!isDelete) {
-			for (int i = 0; i < 9; i++) {
-				for (int j = 0; j < 9; j++) {
-					Cell eachCell = this.getGameBoard().getCellGrid().getCell(i, j);
+	public void enterNumber(Cell currentCell, int number, boolean isHint) {
+
+		/* We don't have to clear empty cell */
+		if (0 != currentCell.getNumber() || 0 != number) {
+
+			/* TODO - this is a tricky part that may require refactoring later */

```

Comment:
I could be completely wrong and might have mistakenly thought I saw
something.
Go ahead and update hintGen. as you like. I am fine as long as it works.

On Tue, Jul 7, 2015 at 7:46 AM, Jeremy Nelson <notifications@github.com>
wrote:

> In src/edu/psu/sweng500/team8/play/GameSession.java
> <https://github.com/garnryang/Team8/pull/10#discussion_r34031127>:
>
> > -			currentCell.clearNumber();
> > -		} else {
> > -			currentCell.setNumber(number);	
> > -		}
> > -		
> > -		/* FIXME - PencilMarkManager updating/wiping out PencilMark number matching entered number */
> > -		if (!isDelete) {
> > -			for (int i = 0; i < 9; i++) {
> > -				for (int j = 0; j < 9; j++) {
> > -					Cell eachCell = this.getGameBoard().getCellGrid().getCell(i, j);
> > +	public void enterNumber(Cell currentCell, int number, boolean isHint) {
> > +
> > +		/* We don't have to clear empty cell */
> > +		if (0 != currentCell.getNumber() || 0 != number) {
> > +
> > +			/* TODO - this is a tricky part that may require refactoring later */
>
> Not sure what you mean by solution check...? Hint generator does not use
> the GameSession class at all.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/garnryang/Team8/pull/10/files#r34031127>.
>